File: arrays/maximumSubarray/maximumSubarray.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int maxSubArray(std::vector<int>& nums) {
        int max_sum = nums[0], curr_sum = 0;
        for (int i = 0; i < nums.size(); ++i) {
            curr_sum += nums[i];
            max_sum = std::max(max_sum, curr_sum);
            if (curr_sum < 0) {
                curr_sum = 0;
            }
        }

        return max_sum;
    }
};

int
main()
{
    Solution A;
    int arr[] = {-1}; 
    int n = sizeof(arr) / sizeof(arr[0]);
    std::vector<int> vec(arr, arr + n);
    std::cout << A.maxSubArray(vec) << std::endl;
}

File: arrays/maximumSubarray/maximumSubarray.py
from typing import List

class Solution(object):
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = nums[0]
        curr_sum = 0
        for i in range(len(nums)):
            curr_sum += nums[i]
            max_sum = max(max_sum, curr_sum)
            if (curr_sum < 0):
                curr_sum = 0
        
        return max_sum
        

A = Solution()
print(A.maxSubArray([-2, 1 ,-3, 4, -1, 2, 1, -5, 4]))